src/common/lib/util/defaults.ts

```diff
@@ -28,7 +28,6 @@ type CompleteDefaults = IDefaults & {
     realtimeRequestTimeout: number;
     recvTimeout: number;
     preferenceConnectTimeout: number;
```

Comment:
Doesn't really make sense to still have a `preferenceConnectTimeout` any more -- can we remove that, and add the two new timeouts you've added (the 10s and 4s) to the (undocumented part of the) clientOptions?